lib/middleware/websocket/controller.rb

```diff
@@ -25,7 +25,11 @@ def on_shutdown(connection)
     end
 
     def on_close(connection)
-      @clients -= find_client(connection: connection)
+      client = find_client(connection: connection)
+      client.first.delete_player
```

Comment:
I don't quite follow what `client.first` is doing for us here.

Reply:
@jgayfer in the `find_client` method you will see that we are using Ruby's `select` function on an array, which returns a list so every time we use `find_client` we need to access the array with `.first` or directly `client[0]`.

The reason it's like this is because we perform `@clients -= client` which only works with two arrays as `arr = arr - [client]` is allowed but `arr = arr - client` does not.

The alternative to this is perhaps having the `find_client` method perform the `.first` operation when it returns so that we always have the `Client` object rather than an array with the `Client` object in it. Then we would have to do something like `@clients -= [client]`

Reply:
@jgayfer see if you prefer the latest commit.

Reply:
I prefer it. Generally a find method returns a singe entity, so this is good.